server/testutil/testfs/testfs.go

```diff
@@ -14,19 +14,44 @@ import (
 
 // MakeTempDir creates and returns an empty directory that exists for the scope
 // of a test.
-func MakeTempDir(t testing.TB) string {
-	tmpDir, err := os.MkdirTemp("", "buildbuddy-test-*")
+func MakeTempDir(t testing.TB, dir string) string {
```

Comment:
It looks like there's three uses of the `dir` argument:

- Most callers will pass `""` to create a brand new temp dir
- One caller passes `/tmp` for creating a socket
- Other callers pass another string to create a child dir within a temp dir returned by `MakeTempDir`

For the sake of making tests easier to write, wdyt about dropping this extra arg? And then for the other use cases:

- For the socket use case, add another func like `TempSocket` which just calls `os.MkdirTemp` and then returns that path joined with `test.sock` (but doesn't actually create the file)
- For the child dir use case, I think we should rely on the cleanup func registered by the `MakeTempDir` to clean up the child dir, and then have a separate function like `testfs.MkdirAll(t testing.TB, rootDir, childPath string)` -- since the convention of the other functions in this package is to take a `rootDir` arg which should be the root test directory returned by `testfs.MakeTempDir`

Reply:
I like these suggestions a lot.